homeassistant/util/dt.py

```diff
@@ -49,6 +49,8 @@ def get_time_zone(time_zone_str: str) -> dt.tzinfo | None:
 
     Async friendly.
     """
+    if not time_zone_str:
```

Comment:
The method signature isn't typed as such.

Reply:
I can switch in onvif device.py itself to provide proper param, but as `zoneinfo.ZoneInfo(None)` is not very safe I thought that it would be better to make `dt_util.get_time_zone` function safer

Reply:
Wouldn't it be better to use an exception handling?

Reply:
@frenck hope I understand you correctly, you meant this `get_time_zone(time_zone_str: str | None = None) -> dt.tzinfo | None:`, right?

@djpremier I didn't want to catch exception as this exception is not what is documented https://docs.python.org/3/library/zoneinfo.html#exceptions-and-warnings so I want to fill a bug in python and I guess it will be changed in the future. But for me doesn't really matter which method we'd use.

Reply:
It's better to have the caller provide the proper argument and keep the parameter type in the signature here as is.

Reply:
check second approach